model/textparse/benchmark_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package textparse
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/prometheus/prometheus/model/exemplar"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/require"
+)
+
+type newParser func([]byte, *labels.SymbolTable) Parser
+
+var newTestParserFns = map[string]newParser{
+	"promtext": NewPromParser,
+	"promproto": func(b []byte, st *labels.SymbolTable) Parser {
+		return NewProtobufParser(b, true, st)
+	},
+	"omtext": func(b []byte, st *labels.SymbolTable) Parser {
+		return NewOpenMetricsParser(b, st, WithOMParserCTSeriesSkipped())
+	},
+}
+
+// BenchmarkParse benchmarks parsing, mimicking how scrape/scrape.go#append use it.
+// Typically used as follows:
+/*
+	export bench=v1 && go test ./model/textparse/... \
+		 -run '^$' -bench '^BenchmarkParse' \
+		 -benchtime 2s -count 6 -cpu 2 -benchmem -timeout 999m \
+	 | tee ${bench}.txt
+*/
+// For profiles, add -memprofile=${bench}.mem.pprof -cpuprofile=${bench}.cpu.pprof
+// options.
+//
+// NOTE(bwplotka): Previous iterations of this benchmark had different cases for isolated
+// Series, Series+Metrics with and without reuse, Series+CT. Those cases are sometimes
+// good to know if you are working on a certain optimization, but it does not
+// make sense to persist such cases for everybody (e.g. for CI one day).
+// For local iteration, feel free to adjust cases/comment out code etc.
+//
+// NOTE(bwplotka): Do not try to conclude "what parser (OM, proto, prom) is the fastest"
+// as the testdata has different amount and type of metrics and features (e.g. exemplars).
+func BenchmarkParse(b *testing.B) {
+	for _, bcase := range []struct {
+		dataFile  string // localized to ./testdata
+		dataProto []byte
+		parser    string
+
+		compareToExpfmtFormat expfmt.FormatType
+	}{
+		{dataFile: "promtestdata.txt", parser: "promtext", compareToExpfmtFormat: expfmt.TypeTextPlain},
+		{dataFile: "promtestdata.nometa.txt", parser: "promtext", compareToExpfmtFormat: expfmt.TypeTextPlain},
+
+		//  We don't pass compareToExpfmtFormat: expfmt.TypeProtoDelim as expfmt does not support GAUGE_HISTOGRAM:
+		// "expfmt.extractSamples: unknown metric family type GAUGE_HISTOGRAM"
+		// TODO(bwplotka): Fix expfmt?
```

Comment:
I'd remove the TODO as it's not a to do for this repository and if we ever make statistics of TODO comments , it's misleading.

```suggestion
```

Reply:
Well, it's our project, so technically it belongs here IF that was a known gap we want to deliver 🤔 

Will replace with concrete issues ID e.g. https://github.com/prometheus/common/issues/430, thanks!